### PR TITLE
Fix service dependency in multi-cloud

### LINKED ISSUE #1237: Fix services dependencies in docker-compose.yaml for deployment
**Issue/Feature Description:**
Gelato installation is failing randomly because of interdependencies of service not handled in docker compose.

Installation Issue - Dependencies not defined

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
The issue is raised to handle smooth and fail safe deployment of gelato services with docker-compose.

**How to reproduce, in case of a bug:**
The issue is random as it depends on order of services getting deployed by docker-compose. 

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**
Note:
The fix is to add:
1. Inter dependencies of services over one other. Use ```depends_on``` feature in docker-compose
2. Add restart policy for each services. It adds up reliability as well.

### PR DESCRIPTION
**Pull request by rajat-soda**
**What type of PR is this?**
Issue Fix PR 

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature

kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:
Fixes the order of service start up so that services do not fail. 

**Which issue(s) this PR fixes**:

Fixes  

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by rajat-soda** [comment]
@AmitRoushan , @asifdxtreme, @PravinRanjan10  Please review .

**Review by asifdxtreme — APPROVED** [review]
LGTM

**Comment by AmitRoushan** [comment]
IMO, The PR need still missed out some services (datamover and dataflow) for datastore dependencies.

**Comment by AmitRoushan** [comment]
IMHO, #1237 issue can only be closed after handling inter dependencies of all services.

**Comment by kumarashit** [comment]
Services has dependency on tidb too. Should we handle that too?

File: docker-compose.yml [hunk 1]
```diff
@@ -142,6 +157,9 @@ services:
       - zookeeper
       - kafka
       - s3
+    depends_on:
```
Comment (AmitRoushan) [inline docker-compose.yml:160]: ```depends_on``` block is already there

**Review by AmitRoushan — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (rajat-soda) [reply docker-compose.yml:160]: > Services has dependency on tidb too. Should we handle that too?

Already handled

**Comment by rajat-soda** [comment]
> IMO, The PR need still missed out some services (datamover and dataflow) for datastore dependencies.

Handled all the services

**Review by AmitRoushan — APPROVED** [review]
LGTM

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**